xernel/kernel/src/sched/scheduler.rs

```diff
@@ -97,50 +107,51 @@ impl Scheduler {
                 }
 
                 while tasks_needed > 0
-                    && !schedulers[j].tasks.is_empty()
-                    && schedulers[j].tasks.len() > avg_tasks
+                    && !schedulers[j].threads.is_empty()
+                    && schedulers[j].threads.len() > avg_tasks
                 {
-                    let task = schedulers[j].tasks.back().unwrap();
+                    let task = schedulers[j].threads.back().unwrap().lock();
 
-                    if task.status == TaskStatus::Running {
+                    if task.status == ThreadStatus::Running {
                         continue;
                     }
 
-                    let task = schedulers[j].tasks.pop_back().unwrap();
+                    drop(task);
```

Comment:
Use the unlock method instead of manually drop. They do the same, but it makes more sense if you use unlock.